pkg/http/responses.go

```diff
@@ -41,15 +42,20 @@ func respond(ctx *fasthttp.RequestCtx, code int, obj []byte) {
 	}
 }
 
-// respondWithETaggedJSON overrides the content-type with application/json and etag header
-func respondWithETaggedJSON(ctx *fasthttp.RequestCtx, code int, obj []byte, etag *string) {
-	respond(ctx, code, obj)
-	ctx.Response.Header.SetContentType(jsonContentTypeHeader)
+// respondWithETag sets etag header
+func respondWithETag(ctx *fasthttp.RequestCtx, etag *string) {
 	if etag != nil {
 		ctx.Response.Header.Set(etagHeader, *etag)
 	}
 }
 
+// respondWithMetadata sets metadata headers
+func respondWithMetadata(ctx *fasthttp.RequestCtx, metadata map[string]string) {
```

Comment:
@artursouza This change generally looks good to me, but on a related note, should the scope of #3172 actually be broader than just the `GetState` API? It looks like `BulkGetState`, `GetSecret` and `BulkGetSecret` should also be responding with metadata in the same fashion (and possibly `OutputBindingMessage` as well?)

Reply:
good catch, I will examine all state related API.

Reply:
LGTM, please fix the same for the other state GET state APIs.

Reply:
I examined `onBulkGetState`, `onGetActorState`, `onGetSecret`, and `onBulkGetSecret`, and found out this support can only applied to State only. This is because `BulkGetSecretResponse`, `GetSecretResponse`, `StateResponse` do not have metadata defined.

I have update the support for `onBulkGetState`, pls. check again.